src/components/Dashboard.jsx

```diff
@@ -27,17 +27,50 @@ const Dashboard = () => {
       isPostOpen: false,
       likeCount: 0,
       isLiked: false,
+      comments: [],
     };
 
     if (postTitle) {
-      postList.push(newPost);
-      localStorage.setItem("posts", JSON.stringify(postList));
+      const newPostList = [...postList, newPost];
+      setPostList(newPostList);
+      localStorage.setItem("posts", JSON.stringify(newPostList));
     }
 
     setPostTitle("");
     textareaRef.current.focus();
   };
 
+  const onSubmitComment = (comment, postid) => {
+    const newComment = {
+      id: new Date().getTime().toString(),
+      author: name,
+      title: comment,
+    };
+    const newPostList = postList.map((post) => {
+      if (post.id === postid) {
+        return { ...post, comments: [...post.comments, newComment] };
+      } else return post;
+    });
+
+    setPostList(newPostList);
+    openCommentBox(postid);
+    localStorage.setItem("posts", JSON.stringify(newPostList));
+  };
+
+  const onDeleteComment = (postId, commentId) => {
+    const newPostList = postList.map((post) => {
+      if (post.id === postId) {
+        const deletedCommentList = post.comments.filter(
+          (comment) => comment.id !== commentId
+        );
+        return { ...post, comments: [...deletedCommentList] };
+      } else return post;
+    });
+
+    setPostList(newPostList);
+    localStorage.setItem("posts", JSON.stringify(newPostList));
+  };
```

Comment:
We can use javascript object reference to make this more simple.
```suggestion
  const onDeleteComment = (postId, commentId) => {
    const _posts = [...postList]
    
    const post = _posts.find(({id}) => id === postId)
    
    if(post === undefined){
        return
    }
    
    post.comments = post.comments.filter(({id}) => id !== commentId)
    
    setPostList(_posts);
    localStorage.setItem("posts", JSON.stringify(_posts));
```